File: blog-common/src/main/java/com/star/common/constant/Result.java
package com.star.common.constant;

import lombok.Data;

import java.io.Serializable;

import static com.star.common.constant.StatusConst.FAIL;
import static com.star.common.constant.StatusConst.SUCCESS;

/**
 * 定义返回结果集
 *
 * @Author: zzStar
 * @Date: 12-16-2020 22:16
 */
@Data
@SuppressWarnings("unchecked")
public class Result<T> implements Serializable {

    /**
     * 返回状态
     */
    private Boolean flag;
    /**
     * 返回码
     */
    private Integer code;
    /**
     * 返回信息
     */
    private String message;
    /**
     * 返回数据
     */
    private T data;

    public static <T> Result<T> success() {
        return restResult(true, null, SUCCESS.getCode(), SUCCESS.getDesc());
    }

    public static <T> Result<T> success(T data) {
        return restResult(true, data, SUCCESS.getCode(), SUCCESS.getDesc());
    }

    public static <T> Result<T> success(T data, String message) {
        return restResult(true, data, SUCCESS.getCode(), message);
    }

    public static <T> Result<T> fail() {
        return restResult(false, null, FAIL.getCode(), FAIL.getDesc());
    }

    public static <T> Result<T> fail(StatusConst statusConst) {
        return restResult(false, null, statusConst.getCode(), statusConst.getDesc());
    }

    public static <T> Result<T> fail(String message) {
        return restResult(false, message);
    }

    public static <T> Result<T> fail(T data) {
        return restResult(false, data, FAIL.getCode(), FAIL.getDesc());
    }

    public static <T> Result<T> fail(T data, String message) {
        return restResult(false, data, FAIL.getCode(), message);
    }

    public static <T> Result<T> fail(Integer code, String message) {
        return restResult(false, null, code, message);
    }

    private static <T> Result<T> restResult(Boolean flag, String message) {
        Result<T> apiResult = new Result<>();
        apiResult.setFlag(flag);
        apiResult.setCode(flag ? SUCCESS.getCode() : FAIL.getCode());
        apiResult.setMessage(message);
        return apiResult;
    }

    private static <T> Result<T> restResult(Boolean flag, T data, Integer code, String message) {
        Result<T> apiResult = new Result<>();
        apiResult.setFlag(flag);
        apiResult.setData(data);
        apiResult.setCode(code);
        apiResult.setMessage(message);
        return apiResult;
    }

}

File: blog-spider/utils/result/result.go
package result

import "fmt"

type RetCode int

const (
	SUCCESS      RetCode = 200
	NotFoundCode RetCode = 404
	FAIL         RetCode = 500
)

var CodeMsg = map[RetCode]string{
	SUCCESS:      "成功",
	NotFoundCode: "资源未找到",
	FAIL:         "失败",
}

type Response struct {
	Flag    bool        `json:"flag"`
	Code    RetCode     `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
}

func (c RetCode) AjaxResult(data interface{}) Response {
	ans := Response{}
	if data != nil {
		ans.Data = data
	}
	if arr, ok := data.([]string); ok && c != SUCCESS {
		ans.Flag = false
		ans.Message = fmt.Sprintf(CodeMsg[c], arr)
		ans.Data = nil
	} else {
		ans.Flag = true
		ans.Message = CodeMsg[c]
	}
	ans.Code = c
	return ans
}
